src/modules/Api.ts

```
@@ -183,6 +192,163 @@ export class Api {
     });
   }
 
+  public async placeCanonicalOrder({
+    order: {
+      side,
+      market,
+      amount,
+      price,
+      makerAccountOwner,
+      expiration = new BigNumber(FOUR_WEEKS_IN_SECONDS),
+      limitFee,
+    },
+    fillOrKill,
+    postOnly,
+    clientId,
+    cancelId,
+    cancelAmountOnRevert,
+  }: {
+    order: {
+      side: ApiSide,
+      market: ApiMarketName,
+      amount: BigNumberable,
+      price: BigNumberable,
+      makerAccountOwner: address,
+      expiration: BigNumberable,
+      limitFee?: BigNumberable,
+    },
+    fillOrKill?: boolean,
+    postOnly?: boolean,
+    clientId?: string,
+    cancelId?: string,
+    cancelAmountOnRevert?: boolean,
+  }): Promise<{ order: ApiOrder }> {
+    const order: SignedCanonicalOrder = await this.createCanonicalOrder({
+      side,
+      market,
+      amount,
+      price,
+      makerAccountOwner,
+      expiration,
+      limitFee,
+      postOnly,
+    });
+
+    return this.submitCanonicalOrder({
+      order,
+      fillOrKill,
+      postOnly,
+      cancelId,
+      clientId,
+      cancelAmountOnRevert,
+    });
+  }
+
+  /**
+   * Creates but does not place a signed canonicalOrder
+   */
+  async createCanonicalOrder({
+    side,
+    market,
+    amount,
+    price,
+    makerAccountOwner,
+    expiration,
+    limitFee,
+    postOnly,
+  }: {
+    side: ApiSide,
+    market: ApiMarketName,
+    amount: BigNumberable,
+    price: BigNumberable,
+    makerAccountOwner: address,
+    expiration: BigNumberable,
+    limitFee?: BigNumberable,
+    postOnly?: boolean,
+  }): Promise<SignedCanonicalOrder> {
+    if (!side || !market) {
```

check that side is actually a valid enum instead of just checking if it's null

---

for example, 'FOO' should throw an error for Side